tech/languages/csharp/csharp-ide.md

```diff
@@ -6,45 +6,36 @@ order: 2
-
-## Visual Studio Code
-
-_.NET Core Only_
-
-Visual Studio Code (aka VS Code) is plugin-based IDE that by itself won't just run C#. You have to install the C# plugin, and NuGet Package Manager.
-
-Download the Visual Studio Code at [code.visualstudio.com](https://code.visualstudio.com) and the C# Extension at [marketplace.visualstudio.com](https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp)
-
-## Eclipse
-
-_.NET Core and Mono_
+First install Eclipse and Eclipse Marketplace
+```
+$ sudo dnf install eclipse eclipse-mpc
+```
+then install the aCute extension from the [Marketplace](https://marketplace.eclipse.org/content/acute-c-edition-eclipse-ide-experimental)
 
-You can use the [aCute plugin](https://marketplace.eclipse.org/content/acute-c-edition-eclipse-ide-experimental) on top of the established and popular [Eclipse IDE](/tools/eclipse/about.html). This C# extension for Eclipse IDE relies on [OmniSharp](http://www.omnisharp.net) and [Language Server Protocol](https://github.com/Microsoft/language-server-protocol). It comes with complete edition features for C# in the Eclipse IDE, relying on the Eclipse IDE provides all typical features of a good IDE (Polyglot development, SCM, deployment...) and give access to a rich ecosystem of extensions for about all existing technologies.
+## Installing MonoDevelop
 
 ```
-$ sudo dnf install eclipse eclipse-mpc
+$ sudo dnf install monodevelop
 ```
-then install the aCute extension from [Eclipse Marketplace](https://marketplace.eclipse.org/content/acute-c-edition-eclipse-ide-experimental)
 
-## MonoDevelop
+## Installing JetBrains Rider
 
-_Mono only_
+Apply for Open Source [license](https://www.jetbrains.com/store/?fromMenu#edition=discounts) and download Rider at [jetbrains.com/rider](http://jetbrains.com/rider) or download the [Early Access](https://www.jetbrains.com/rider/eap) version.
 
-MonoDevelop provides the best C# experience on Mono, the open-source multiplatform implementation of the .NET Framework.
+## Installing Visual Studio Code
 
-[monodevelop.com](http://www.monodevelop.com)
+Download the Visual Studio Code at [code.visualstudio.com](https://code.visualstudio.com) and the C# Extension at [marketplace.visualstudio.com](https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp)
 
-```
-$ sudo dnf install monodevelop
-```
 
 ## Summary
 
-What would we recommend? JetBrains Rider is the best C# IDE, however it is not open source. VS Code is great for .NET Core and MonoDevelop for Mono. C# in Eclipse IDE is still young and not very friendly to former Windows developer, but it can be useful if you're already an Eclipse IDE user.
+JetBrains Rider is the most complete C# IDE, however it is not open source. VS Code and MonoDevelop work well for .NET Core and for Mono, respectively. C# via Eclipse is still young and not very friendly for the former Windows developer, but it can be useful if you're already an Eclipse IDE user.
```

Comment:
> Code and MonoDevelop work well for...

 - how about a `with` instead of `for`?

Reply:
Could we split it into multiple blocks / separate lines at least, so we could distinguish recommendations for different IDEs and not have 'one single block?

Reply:
poke @aslicerh

Reply:
@pvalena 
I'd be fine with breaking it out into three separate sentences.

Regarding `with` vs `for`, I, personally like `for`, as in the context:
`VS Code and MonoDevelop work well for [developing] .NET Core and Mono, respectively.`
I.E., actually discussing what they VS Code and MonoDevelop are used for.

Saying they work well *with* .NET Core and Mono implies (to me at least) that VS Code and MonoDevelop have a use independent of .NET and Mono that happens to align with .NET and Mono (when that use is quite purposeful and designed for these languages).

So, now that I have over-analyzed this discussion, let me say that I don't really have strong feelings one way or the other. I think `for` sounds better, but I don't really think `with` is incorrect. So I'll go with the consensus.

Reply:
> I'd be fine with breaking it out into three separate sentences.

Please do, thanks!

> Regarding with vs for,

AFAIK it's not usual to use `for` in this context. You use `with` with unanimate objects; and  `for` would IMHO suggest the object is somehow 'alive'.
Anyway, I will ask on my English lessons :)

*[ non-blocking ]*